dart-mode.el

```diff
@@ -562,6 +579,251 @@ This uses `dart-format-path' to find the formatter."
       (dart-format-region beg (point)))))
 
 
+;;; Dart analysis server
+
+(defvar dart-debug nil)
+(defvar dart-enable-analysis-server nil)
+(defvar dart-analysis-server nil)
+(defvar dart-analysis-server-snapshot-path
+  "/usr/lib/dart/bin/snapshots/analysis_server.dart.snapshot")
+(defvar dart-analysis-roots nil)
```

Comment:
This, `dart-analysis-server-next-id`, and `dart-analysis-server-callbacks` should be private.